packetery/src/Packetery/Checkout.php

```diff
@@ -442,4 +452,36 @@ private function getChosenMethod(): string {
 
 		return '';
 	}
+
+	/**
+	 * Gets carrier id from chosen shipping method.
+	 *
+	 * @param string $chosenMethod Chosen shipping method.
+	 *
+	 * @return string|null
+	 */
+	private function getCarrierId( string $chosenMethod ): ?string {
+		$carrierId = null;
+		$matches   = [];
+		if ( preg_match( '/^' . self::CARRIER_PREFIX . '(\w+)$/', $chosenMethod, $matches ) ) {
```

Comment:
Prosím o nepoužívání pregmatchů. Použij explode() a array_pop(). Také jsem si všiml, že podobný pregmatch je použit na jiném místě. Chci aby extrakce carrierId byla na jednom místě. Možná vytvořit getExternalCarrierId() : ?int a zrefaktorovat isPickupPointMethod()